xml/core_registers.xml

```diff
@@ -70,7 +70,48 @@ same project unless stated otherwise.
             available version of this spec.
             </value>
         </field>
-        <field name="0" bits="27:18" access="R" reset="0" />
+        <field name="0" bits="27" access="R" reset="0" />
+        <field name="extcause" bits="26:24" access="R" reset="0">
+            When {dcsr-cause} is 7, this optional field contains the value of a
+            more specific halt reason than "other." Otherwise it contains 0.
+
+            <value v="0" name="critical error">
+                The hart entered a critical error state. Part of the
+                ((Smdbltrp)) extension.
+            </value>
+
+            All other values are reserved for future versions of this spec, or
+            for use by other RISC-V extensions.
+        </field>
+        <field name="0" bits="23:20" access="R" reset="0" />
+        <field name="cetrig" bits="19" access="R/W" reset="0">
+            This bit is part of ((Smdbltrp)) and only exists when that extension
+            is implemented.
+
+            <value v="0" name="disabled">
+                A hart in a critical error state does not enter Debug Mode but
+                instead asserts the critical-error signal to the platform.
+            </value>
+
+            <value v="1" name="enabled">
+                A hart in a critical error state enters Debug Mode instead of
+                asserting the critical-error signal to the platform. Upon such
+                entry into Debug Mode, the cause field is set to 7, and the
+                extcause field is set to 0, indicating a critical error
+                triggered the Debug Mode entry.  This cause has the highest
+                priority among all reasons for entering Debug Mode. Resuming
+                from Debug Mode following an entry from the critical error state
+                returns the hart to the critical error state.
+            </value>
+
+            [NOTE]
+            ====
+                When {dcsr-cetrig} is enabled, resuming from Debug Mode
+                following an entry due to a critical error will result in an
+                immediate re-entry into Debug Mode due to the critical error.
```

Comment:
The debugger may resume with cetrig set to 0 to allow the platform defined actions on critical-error signal to occur. Other possible actions include initiating a hart or platform reset using the Debug Module reset control.

Reply:
Adding this.

Reply:
Thanks!